kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -47,18 +47,51 @@
 
 _LOGGER: Final = logging.getLogger(__name__)
 
+
+class CustomStep:
+    """Encapsulates a custom step definition consisting of an abstract pattern and its execution function."""
+
+    pattern: KSequence
+    exec_fn: Callable[[Subst, CTerm, CTermSymbolic], KCFGExtendResult | None]
```

Comment:
You can turn this into a method.

Reply:
I guess probably not, but you can consider extracting another class for it.

Reply:
Added a new type alias in https://github.com/runtimeverification/evm-semantics/pull/2698/commits/012224537ccaaa948f06b58132bb193cbbe0a3ac.